pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -106,7 +106,7 @@ func (t ResolvedPipelineTask) IsCustomTask() bool {
 
 // IsMatrixed return true if the PipelineTask has a Matrix.
 func (t ResolvedPipelineTask) IsMatrixed() bool {
-	return t.PipelineTask.Matrix != nil && len(t.PipelineTask.Matrix.Params) > 0
+	return t.PipelineTask.Matrix.HasParams() || t.PipelineTask.Matrix.HasInclude()
```

Comment:
this definition is different from pt.IsMatrixed in v1beta1/v1 -- also not sure that we need to redefine the function here

```suggestion
	return t.PipelineTask.IsMatrixed()
```